api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java

```diff
@@ -20,10 +21,12 @@ public Object pullDataFromParent(String category, Map<String, String> resourceLi
         String preferredClient = Context.getAdministrationService().getGlobalProperty(RESOURCE_PREFERRED_CLIENT);
         String url = address + resourceLinks.get(preferredClient);
 
+        ClientFactory clientFactory = new ClientFactory();
         if (REST_CLIENT_KEY.equals(preferredClient)) {
```

Comment:
It is a redundant condition.  You use the same if statement in the ClientFactory. IMHO you can use the createClient method only once with parameter "preferredClient"

Reply:
We need some kind of client interface. If Karol didn't use this condition he wouldn't be able to know how to cast the object returned by createClient(...) method.